src/main/java/core/basesyntax/dao/FruitDaoImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import java.util.List;
+
+public class FruitDaoImpl implements FruitDao {
+    @Override
+    public void add(Fruit fruit) {
+        Storage.fruits.add(fruit);
+    }
+
+    @Override
+    public Fruit getFruitByName(String fruitName) {
+        return Storage.fruits.stream()
+                .filter(f -> f.getName().equals(fruitName))
+                .findFirst()
+                .orElseGet(() -> new Fruit(fruitName));
+    }
```

Comment:
Returning a new Fruit object when one with the specified name doesn't exist can be misleading. It would be better to return an Optional<Fruit> to explicitly handle the case where the fruit is not found.

Reply:
++